crates/astria-cli/src/sequencer/account.rs

```diff
@@ -25,20 +24,64 @@ impl Command {
             SubCommand::Create(create) => create.run(),
             SubCommand::Balance(balance) => balance.run().await,
             SubCommand::Nonce(nonce) => nonce.run().await,
+            SubCommand::Recover(recover) => recover.run(),
         }
     }
 }
 
 #[derive(Debug, Subcommand)]
 enum SubCommand {
-    /// Generates a new ED25519 keypair.
+    /// Generates a new Ed25519 keypair.
     Create(Create),
+    /// Recovers an Ed25519 keypair from a mnemonic.
+    Recover(Recover),
     /// Queries the Sequencer for the balances of an account.
     Balance(Balance),
     /// Queries the Sequencer for the current nonce of an account.
     Nonce(Nonce),
 }
 
+#[derive(Debug, clap::Args)]
+struct Recover {
+    /// The address prefix
+    #[arg(long, default_value = "astria")]
+    prefix: String,
+    /// The recovery mnemonic phrase, must be wrapped in quotes.
+    #[arg(long)]
+    mnemonic: String,
+}
+
+impl Recover {
+    fn run(self) -> eyre::Result<()> {
+        let bip39_mnemonic = bip39::Mnemonic::from_phrase(&self.mnemonic, bip39::Language::English)
+            .wrap_err("failed to recover mnemonic from phrase")?;
+        let seed = bip39::Seed::new(&bip39_mnemonic, "");
+        let seed_bytes: [u8; 32] = seed.as_bytes()[0..32]
+            .try_into()
+            .wrap_err("failed to convert seed to 32 bytes")?;
+
+        let signing_key = SigningKey::from(seed_bytes);
+
+        let pretty_signing_key = hex::encode(signing_key.as_bytes());
+        let pretty_verifying_key = hex::encode(signing_key.verification_key().as_bytes());
+
+        let pretty_address: Address = Address::builder()
+            .array(signing_key.address_bytes())
+            .prefix(&self.prefix)
+            .try_build()
+            .wrap_err("failed constructing a valid from address from the provided prefix")?;
+
+        // TODO: don't print private keys to CLI, prefer writing to file:
+        // https://github.com/astriaorg/astria/issues/594
+        println!(
+            "Recover Sequencer Account\n\nPrivate Key: {pretty_signing_key}\nPublic Key:  \
```

Comment:
As below, I wouldn't do the manual spacing here. Let's fix this in a followup by providing a mechanism to fix this across all outputs (e.g. so that adding a character or changing the variable output doesn't require manually re-indenting things).